### ISSUE TITLE
Graphical glitch when using some shortcuts in Geany Preview

### ISSUE THREAD

**Issue by ralf3u** [xiota/geanylua-scripts#9]
When using some special shortcuts (Toggle editor 1, Toggle editor 2 and both Switch Views), then one can see the scrollbar of the Preview for a part of the second somewhere else.

If for example the sidebar is on the right side, and if the widths of editor and sidebar are about 3:1 (see image 1), then, after having pressed some shortcuts, the sidebar is moving to the left side (see image 2), so that the scrollbar of the Preview is more on the left side, and then the content expands to the right side (see image 3). The images are screenshots from a video that I did with a digital camera.

image 1
![1](https://github.com/xiota/geanylua-scripts/assets/68117017/ea312ed8-7825-43e9-9296-77b8de0ab2fa)

image 2
![2](https://github.com/xiota/geanylua-scripts/assets/68117017/874e0940-d90e-41cb-8c0e-7d0d27d351ef)

image 3
![3](https://github.com/xiota/geanylua-scripts/assets/68117017/7dd3f96b-e73a-4fe6-aafc-bc475cad43a3)

History
The glitch was mentioned for the first time in https://github.com/xiota/geanylua-scripts/issues/4#issuecomment-1152776068 in point 3, and later in https://github.com/xiota/geanylua-scripts/issues/7#issuecomment-1451950479.

Environment
Lubuntu 23.04 with Geany 1.38.

Remark
Because you wrote that you don't see the graphical glitch at all (see https://github.com/xiota/geanylua-scripts/issues/7#issuecomment-1460026893):
Maybe it would be good if you notice your environment like CPU, RAM, distribution name, Geany version number, and graphic card, so if someone feels too much disturbed by the graphical glitch, there is at least the possibility to change the environment. And maybe the values could help to find the cause of the issue.

**Comment by xiota** [comment]
I doubt most people would see or be bothered by the glitch because normally people don't hide and show the sidebar as much as you do.  What GPU are you using?  Have you tried X11 vs Wayland?  Does the glitch occur with different window managers or desktop environments?

**Comment by ralf3u** [comment]
> What GPU are you using?

Intel UHD Graphics

>Have you tried X11 vs Wayland? Does the glitch occur with different window managers or desktop environments?

I will try Wayland with a live-system of Ubuntu. I'm still using Lubuntu, so only LXQt and Openbox.

**Comment by xiota** [comment]
I do remember seeing a glitch when I rapidly hid and showed the sidebar.  Otherwise it wasn't noticeable.  Try KDE Plasma too.

**Comment by ralf3u** [comment]
OK, I intend to try Ubuntu and Kubuntu, both as live-system. Is live-system OK, or should I install both systems?

**Comment by xiota** [comment]
Try live first because it's easier.  Sometimes live environments use "safe" drivers that may not perform as well.  So you might want to install the distro if you didn't see the results you want.  If you have a machine set up for distro testing, you could try Endeavour and Manjaro.  They use a different package manager from Ubuntu, so installation would be different.

**Comment by ralf3u** [comment]
I can still see the graphical glitch in Geany 2.0.

**Comment by ralf3u** [comment]
I just want to document that on Lubuntu 24.04 with Geany 2.0 with eink-display or TFT-display, there is still the graphical glitch.

**Comment by ralf3u** [comment]
I think that there are two possibilities to show "only sidebar":
- disable editor
- maximize sidebar

In "disable editor" the separation line between editor and sidebar can't be grabbed on one side of the window because there is no separation line any more.
In "maximize sidebar" there is still a separation line on one side of the window that can be grabbed.

This issue is using "disable editor" in the shortcut because I can't grab a separation line on one window-side. What about the idea to try "maximize sidebar" as shortcut to see if the graphical glitch is still there?

**Comment by xiota** [comment]
"Maximize" sidebar would be troublesome.  Would require saving current size, changing it, restoring it.  I had a plugin that tried to do that to save different sidebar sizes for maximized and unmaximized windows.  It got confused some small percentage the time (maybe around 5-10%).  Then one day, it stopped working.  I didn't bother to investigate the reason.  Possibly some GTK or Geany change prevents plugins from changing sidebar size.

**Comment by xiota** [comment]
I wonder whether the glitch (second screenshot) could be caused or mitigated by some animation effect.

**Comment by ralf3u** [comment]
>I wonder whether the glitch (second screenshot) could be caused or mitigated by some animation effect.

I think that the change is just slow. The sidebar moves to the left. And then the sidebar will be extended to the right. If both steps would be executed much faster, then maybe one would not see the graphical glitch.

**Comment by ralf3u** [comment]
That is really strange:
If I deactivate the plugin `Preview` and the plugin `Xi/Tweaks`, if after restarting Geany I can see editor&sidebar while in the sidebar in the tab `Documents` there is a scrollbar, then if I press the geanylua- shortcut `toggle-editor-1` then I can't see at all a graphical glitch.

Do you have an explication?

**Comment by ralf3u** [comment]
>That is really strange:
If I deactivate the plugin Preview and the plugin Xi/Tweaks, if after restarting Geany I can see editor&sidebar while in the sidebar in the tab Documents there is a scrollbar, then if I press the geanylua- shortcut toggle-editor-1 then I can't see at all a graphical glitch.

It seems to be that geany-lua is not the problem, but preview.
Should I close this issue and open a new issue at preview?

**Comment by xiota** [comment]
>  Should I close this issue and open a new issue at preview?

No.  It's probably the webbrowser component is relatively slow to load.  There's nothing I can do about it.

**Comment by ralf3u** [comment]
Because I can't reproduce the issue when the plugin `Preview` is deactivated, the issue should be assigned to the plugin `Preview`. I close this issue here at geanylua and I will open today a new issue at preview.

>It's probably the webbrowser component is relatively slow to load. There's nothing I can do about it.

If you can't do nothing about it, you can close the issue at preview.

**Comment by ralf3u** [comment]
I reopen this issue because of https://github.com/xiota/geany-preview/issues/61#issuecomment-2691653097.

**Comment by ralf3u** [comment]
Yesterday I opened at Geany issue 4251. And today I opened at Geany issue 4252.

**Comment by ralf3u** [comment]
Today I had an interesting idea how to deal with the memory of the sidebar width: When 
sidebar width=100% then the memory should not remember this sidebar width. I mentioned that some hours ago at 4251 and 4252 at geany.

**Comment by ralf3u** [comment]
Yesterday I opened at Geany issue 4254. And today I opened at Geany issue 4255. And I have the intention to open some more issues at geany the next days.

**Comment by ralf3u** [comment]
FYI, some days ago I had the idea not use switch-views of geanylua-scripts any more because of this issue and the focus issue (see issue number 10). Today I realized this idea, so I use in geanylua only toggle-menu-bar. It was a good idea because that lead me to open the issues 4251 and 4252 at geany some days ago. And then I recognized that I can use the Geany shortcut "Toggle Sidebar" as "Toggle between only editor and only sidebar" (this is for me the most important shortcut for the plugin Preview) by maximizing the sidebar width. That lead me to open three further issues at geany, 4254, 4255 and 4256 the last days. When I use nowadays the Geany shortcut "Toggle Sidebar", then I have exactly the same behaviour like the shortcut "switch-views-(editor-↔-sidebar)" of switch-views, but without the graphical glitch issue. Unfortunately the focus issue is the same.

**Comment by xiota** [comment]
I wouldn't expect 4254 or 4256 to be implemented because geany is primarily a text editor.  Changing sidebar to 100% would not be expected behavior.  Some projects (KDE, gnome, Firefox) would even add code to prevent it.  (Try with Dolphin file manager and Firefox.)

**Comment by ralf3u** [comment]
> Changing sidebar to 100% would not be expected behavior. [...] Try with Dolphin file manager and Firefox.

Just because some go one direction doesn't mean Geany will do the same.

**Comment by xiota** [comment]
I'm pointing out that it's a use case that is unlikely to be implemented.  Doing so potentially causes problems for typical users.  So by even opening the issue, you actually risk some developer blocking your ability to change the sidebar size to 100%, as has been done with other projects.  I don't expect geany devs to do that, but have seen other projects implement the opposite of what a user asked for.

**Comment by ralf3u** [comment]
>I wouldn't expect 4254 or 4256 to be implemented [...]

Your sentence inspired me to open a new issue at Geany. The issue number is 4257. In case 4251 will not be be realized, 4257 is the backdoor solution.

**Comment by ralf3u** [comment]
The cause of the graphical glitch is located in Preview (see issue number 89 at xiota/geany-preview).
That is the reason why I'm closing this issue.
